### DISCUSSION TITLE
Stuck on Installing Script Kit

### DISCUSSION THREAD

**Discussion by indrasvat** [johnlindquist/kit#769] [Errors]
Kit suddenly stopped responding to the `⌘+;` shortcut. I was unable to get it to show up, even though I could see the process running.

So, I blew away `~/.kit`, killed the process, deleted the binary and re-installed it. The `~/.kenv` was untouched.

Now, after starting Kit, it seems to be stuck on the **Installing Script Kit** step.

```bash
$ ps -o etime,pid $(pgrep -x Kit)

ELAPSED   PID
29:24     77870
```

Kit version: `1.17.2`. I'm on Apple M1/Monterey.

<img width="880" alt="Screen Shot 2022-05-11 at 10 44 13 AM" src="https://user-images.githubusercontent.com/817245/167913370-1021fd3c-429e-4318-b094-1e7d4883d435.png">

How can I troubleshoot this?

**Comment by indrasvat** [comment]
I've tried this a couple of times now. It always gets stuck installing. Not sure how to proceed.

**Reply by johnlindquist** [reply]
Would you love to have you over on our new discord server to troubleshoot it: https://discord.gg/uyKBCGR2

Otherwise, try force-quitting while it's stuck (just long enough for the ~/.kit dir to be created/extracted), then in the terminal run:

```
cd ~/.kit
PATH="./node/bin/:$PATH" npm i
```

That should use the bundled version of npm to install the required packages. I imagine something is failing there...

"Installing" is essentially:
1. Extract the version locked/bundled https://github.com/johnlindquist/kit to ~/.kit
2. Extract the version locked/bundled version of node
3. npm install on ~/.kit
4. Run a couple small config/setup task in ~/.kit/setup

**Reply by indrasvat** [reply]
Thanks, @johnlindquist!

On the version of `node`:

```bash
$ ls -Altr `which node` && node -v
lrwxr-xr-x  1 user  staff    40B Mar 18 12:08 ~/user/.volta/bin/node@ -> ~/user/.volta/bin/volta-shim

v18.1.0
```

I'll try the above steps as well.

**Reply by johnlindquist** [reply]
@kodeninja Sorry, I was trying to explain that there's a version of node including here:

`~/.kit/node`

That's why we temporarily set the `PATH=./node/bin/:$PATH`  in the `npm i` command.

Your system's node version shouldn't matter. Kit does not expect node to be installed by the user, that's why it bundles it.

**Reply by indrasvat** [reply]
@johnlindquist, `PATH="./node/bin/:$PATH" npm i` worked!

I think I know what the issue is. I have my global npm registry set to our _internal_, corporate registry. And to access that, you have to be on the VPN.

I _think_ I was _not_ on the VPN whenever I happened to be trying to re-install Kit 🤦🏼‍♂️ 

Perhaps it makes sense to ship with an `.npmrc` as well, as part of the install process, so that Kit can always use the public NPM registry 🤷🏼‍♂️. Not sure if that makes sense, or if it's even the issue.

**Reply by johnlindquist** [reply]
Yeah, that makes total sense. I actually intended to add an .npmrc with the previous build, but forgot to commit it 🤦‍♂️

Glad you got it working.

Let me know if there's anything else I can help with. Or if you want to chat script ideas, there's always the discord 😇

https://discord.gg/44smSDRT

**Reply by indrasvat** [reply]
Thanks for all the help, @johnlindquist. I'll make sure to post on the Discord server from now on 😀
